src/script/run-make.sh

```diff
@@ -72,6 +72,7 @@ Note that the binaries produced by this script do not contain correct time
 and git version information, which may make them unsuitable for debugging
 and production use.
 EOM
+    test -d $HOME/.ccache || mkdir $HOME/.ccache
```

Comment:
this changes the behavior of ccache. as `$HOME/.ccache` has the highest priority.

but i think a better fix is not to `mkdir -p $HOME/.ccache`. instead, we should respect users's preferences:

1. save current `XDG_CACHE_HOME` to a temporary variable before setting it if this variable is not set
2. set it with `XDG_CACHE_HOME=$top_srcdir/install-deps-cache`
3. after done with loading wheels restore the `XDG_CACHE_HOME` if it was set, `reset` it otherwise

Reply:
Yes, we needs a clear path instead of conditional choice. But most of the machines use $HOME/.ccache, do we need to be consistent?

Reply:
> we needs a clear path instead of conditional choice. 

i am not sure i understand what you meant "a clear path instead of conditional choice".

> But most of the make check machines use $HOME/.ccache

please be more specific. why? and how does this address my concern on human users?

Reply:
sorry for delay,  I mean the default ccache cache_dir is a  conditional choice
```
cache_dir (CCACHE_DIR)
This option specifies where ccache will keep its cached compiler outputs.
On non-Windows systems, the default is $HOME/.ccache if such a directory exists,
otherwise $XDG_CACHE_HOME/ccache if XDG_CACHE_HOME is set,
otherwise $HOME/Library/Caches/ccache (macOS) or $HOME/.config/ccache (other systems)
```
A clear path means we need to specify a path, /home/jenkins-build/.ccache ($HOME/.ccache) maybe a good choice, as the status is that most of the make check machines use /home/jenkins-build/.ccache ($HOME/.ccache)，I think the dir /home/jenkins-build/.ccache ($HOME/.ccache) is created manully before.

Last few days, some new nodes add to x86 make check, like https://jenkins.ceph.com/job/ceph-pull-requests/133296/consoleFull, ccache always missed.

Reply:
again, i think there are two user of this script: build machine and human developers. i'd respect user's option. and a solution was proposed to cater both needs. am i missing anything?

Reply:
oh, my bad, I ignore the human developers, all my thought is  about the build machine.  I will try to modify.